service/controller/v1/resource/chart/spec.go

```diff
@@ -21,3 +21,11 @@ type ChartState struct {
 	// e.g. 0.1.2
 	ReleaseVersion string
 }
+
+// Equals asseses the equality of ChartStates with regards to distinguishing fields
+func (a *ChartState) Equals(b ChartState) bool {
```

Comment:
Didn't find a better place to put the `Equals` method. Feedback welcome.

Reply:
I think it's ok to have it here close to `ChartState` type definition :+1: